modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/edit_form_instance.jsp

```diff
@@ -172,6 +167,8 @@ renderResponse.setTitle((formInstance == null) ? LanguageUtil.get(request, "new-
 						"fieldTypes", ddmFormAdminDisplayContext.getDDMFormFieldTypesJSONArray()
 					).put(
 						"formInstanceId", formInstanceId
+					).put(
+						"formSettingsContext", ddmFormAdminDisplayContext.getDDMFormSettingsContext(pageContext)
```

Comment:
I see no direct relation between the `App.es.js` and the `formSettingsContext`, isolated the  settings logic into another component would help on Form's maintenance by reducing the code complexity into small pieces.

Reply:
@miltonmc, After removing the Modal from the JSP, it was necessary to pass the context to the modal that is in react and this is the best way to pass the entire form settings context to the component that is inside FormBuilder.